Userland/Libraries/LibWeb/URL/URL.cpp

```diff
@@ -535,6 +539,31 @@ bool host_is_domain(AK::URL::Host const& host)
     return host.has<String>() && host.get<String>() != String {};
 }
 
+// https://url.spec.whatwg.org/#potentially-strip-trailing-spaces-from-an-opaque-path
+WebIDL::ExceptionOr<void> strip_trailing_spaces_from_an_opaque_path(URL& url)
+{
+    auto& vm = url.realm().vm();
+
+    // 1. If url’s URL does not have an opaque path, then return.
+    if (!url.has_opaque_path())
+        return {};
+
+    // 2. If url’s URL’s fragment is non-null, then return.
+    if (url.fragment().has_value())
+        return {};
+
+    // 3. If url’s URL’s query is non-null, then return.
+    if (url.query().has_value())
+        return {};
+
+    // 4. Remove all trailing U+0020 SPACE code points from url’s URL’s path.
```

Comment:
This spec step states we should be trimming space code units, not ascii whitespace - also, we don't need to return failure here on OOM, we can just `MUST` the result and crash instead if we run out of memory.